deltachat-jsonrpc/src/api/mod.rs

```diff
@@ -1848,6 +1848,22 @@ impl CommandApi {
         }
     }
 
+    async fn send_draft(&self, account_id: u32, chat_id: u32) -> Result<u32> {
```

Comment:
I did it this way because it was the simplest way to implement it with my limited Rust copy+paste skills, but probably it is more bug-free to pass the draft/msg id instead to be sure you are sending the draft you are expecting to send and not another draft that replaced the draft you think you are sending in the mean time

anyway for bots this is good enough, desktop is maybe more sensitive, feel free to pick up this pull request and improve

Reply:
I would be for providing just msg id of draft and the function tries to load it and if it can then it sends it.

Reply:
also if the api is not final you can make it start with `misc_`. BTW the current api for setting drafts is also not quite ready as it does not return the message id of the draft yet and has no option to keep the same msg id even when you change the text it will create a new message.

Reply:
the danger of an just send whatever draft function is that it could produce race conditions like:
1. create draft
2. send draft because user clicked on send
3. send draft because user clicked on send multiple times (this one could be delayed for some reason
4. user creates new draft for new message
5. Unfinished message is send due to the delayed send draft call

Admittedly this is theoretically, but I still think a api where you create a draft then send it by it's id is better.
Anyways we should really think about this, also how to implement it nicely in desktop.

Reply:
yes, it absolutely makes sense to pass msg id, but if no one will take care of this PR soon, I would like to have this api at least for bots, where that is not a problem, to at least have something to work with in bots